### ISSUE TITLE
reconnect when password= supplied asks user for password already supplied

### ISSUE THREAD

**Issue by spacelama** [novnc/noVNC#925]
As mentioned in https://github.com/novnc/noVNC/pull/747#issuecomment-317798663 but retracted perhaps erroneously later, the password isn't set on reconnects when supplied as a cgi parameter.

I think I can see why in that setPassword sets UI.reconnect_password, and future reconnects depend on UI.reconnect_password, but supplying password as cgi parameter doesn't set UI.reconnect_password.  This does not achieve the stated aim, but I can't see how to do it correctly:
```
--- a/app/ui.js
+++ b/app/ui.js
@@ -1039,6 +1039,7 @@ var UI = {
 
         if (typeof password === 'undefined') {
             password = WebUtil.getConfigVar('password');
+            UI.reconnect_password = password;
         }
 
         if (password === null) {
```

**Comment by samhed** [comment]
Yes, I don't know why he said that it wasn't actually a problem. I can reproduce the issue indeed.

**Comment by samhed** [comment]
The diff you posted above is precisely the solution to this issue. What do you mean it "does not achieve the stated aim"?

**Comment by samhed** [comment]
Fixed in 2e735160bf89d3fe23034b03cd9dfc0e1876d83a

**Comment by spacelama** [comment]
Dunno!  Expect Heisenbug in my and 747's testing regime.  Works For Me™ Now.
